### ISSUE TITLE
[feat/question] block scope separation

### ISSUE THREAD

**Issue by eLeontev** [airbnb/javascript#2251]
Inconsistency between good/bad samples in block/scope area.

In original section https://github.com/airbnb/javascript#whitespace--after-blocks mentioned:
we need to follow empty line between/after the block.

In different areas also mentioned next cases as good:
https://github.com/airbnb/javascript#references--disallow-var

```javascript
// good, use the let.
let count = 1;
if (true) {
  count += 1;
}
```
or
https://github.com/airbnb/javascript#functions--note-on-blocks

```javascript
// good
let test;
if (currentUser) {
  test = () => {
    console.log('Yup.');
  };
}
``` 
and other

**Could you clarify and align the docs about do we need to split variables related only to the block?**

```javascript
// good or bad
const condition = true;
if (condition) {
  //do condition 
}

// good or bad
const condition = true;

if (condition) {
  //do condition 
}
``` 
It is minore but anyway stylegude should cover it

**Comment by ljharb** [comment]
I don't think there needs to be a proscription here; I'd do whatever ends up being clearest. In my experience, that's usually "right next to the block", but I don't see any issue with lacking a rule here.

**Comment by eLeontev** [comment]
Yep, it is not the big deal... just to be clear, the codestyles main goal is to be able to answer to all style questions like "how to write it" to simpleify review process to link to it.
In my case I faced with  question above... and yes we discussed and closed our own case, but it was strange to have no option to link to styles.  

In other words the answer `I don't see any issue` is not appropriate for me (sorry  :)). 


Once correct behavior will be agreed I could add this rule to styles.
